src/foam/util/UIdGenerator.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.util',
+  name: 'UIDGenerator',
+  flags: ['java'],
+
+  javaImports: [
+    'java.util.concurrent.ThreadLocalRandom',
+    'java.util.ArrayList',
+    'java.util.Random'
+  ],
+
+  properties: [
+    {
+      name: 'seqNo',
+      class: 'Int',
+      javaFactory: 'return 0;'
+    },
+    {
+      name: 'lastTimeCalled',
+      class: 'Long',
+      javaFactory: 'return System.currentTimeMillis() / 1000;'
+    },
+    {
+      name: 'randInts',
+      class: 'List',
+      javaType: 'ArrayList<Integer>',
+      javaFactory: `
+        Random r = new Random();
+        ArrayList<Integer> randInts =  new ArrayList<>();
+        long seed = 1;
+        r.setSeed(seed);
+        int lowerbound = 12;
+        int upperbound = 30;
+        for ( int i = 0; i < upperbound; i++ )
+        {
+          randInts.add(r.nextInt(Math.max(i, lowerbound - 1)));
```

Comment:
How is it ensured that an random index isn't re-used?

Reply:
An random index might be re-used.